fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeMetastore.java

```diff
@@ -63,10 +65,15 @@ public DeltaLakeMetastore(String catalogName, IMetastore metastore, Configuratio
         this.delegate = metastore;
         this.hdfsConfiguration = hdfsConfiguration;
         this.properties = properties;
+        long checkpointCacheSize = Math.round(Runtime.getRuntime().maxMemory() *
+                properties.getDeltaLakeCheckpointMetaCacheMemoryUsageRatio());
+        long jsonCacheSize = Math.round(Runtime.getRuntime().maxMemory() *
+                properties.getDeltaLakeJsonMetaCacheMemoryUsageRatio());
 
         this.checkpointCache = CacheBuilder.newBuilder()
                 .expireAfterWrite(properties.getDeltaLakeCheckpointMetaCacheTtlSec(), TimeUnit.SECONDS)
-                .maximumSize(properties.getDeltaLakeCheckpointMetaCacheMaxNum())
+                .weigher((key, value) -> Math.toIntExact(SizeEstimator.estimate(key) + SizeEstimator.estimate(value)))
```

Comment:
This interface estimate is not accurate and the possible error is relatively large. cc @gengjun-git

Reply:
You can remove the memory tracker code, I will add it in another PR https://github.com/StarRocks/starrocks/pull/49841